app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt

```diff
@@ -18,25 +18,67 @@
 
 package org.kiwix.kiwixmobile.main
 
+import android.content.Intent
 import android.os.Bundle
+import androidx.appcompat.view.ActionMode
+import androidx.navigation.NavController
+import androidx.navigation.findNavController
+import androidx.navigation.ui.AppBarConfiguration
+import androidx.navigation.ui.setupWithNavController
+import kotlinx.android.synthetic.main.activity_new_navigation.nav_view
 import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.core.base.BaseFragmentActivityExtensions
 import org.kiwix.kiwixmobile.core.di.components.CoreComponent
 import org.kiwix.kiwixmobile.core.main.CoreMainActivity
 import org.kiwix.kiwixmobile.kiwixActivityComponent
 
 class KiwixMainActivity : CoreMainActivity() {
+  private lateinit var navController: NavController
+  private lateinit var appBarConfiguration: AppBarConfiguration
+  private var actionMode: ActionMode? = null
+
   override fun injection(coreComponent: CoreComponent) {
     kiwixActivityComponent.inject(this)
   }
 
+  private val finishActionModeOnDestinationChange =
+    NavController.OnDestinationChangedListener { controller, destination, arguments ->
+      actionMode?.finish()
+    }
+
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
-    setContentView(R.layout.activity_kiwix_new_navigation)
+    setContentView(R.layout.activity_new_navigation)
+
+    navController = findNavController(R.id.nav_host_fragment)
+    navController.addOnDestinationChangedListener(finishActionModeOnDestinationChange)
+    appBarConfiguration = AppBarConfiguration(navController.graph)
```

Comment:
this is only assigned, never used. It needs to be used in `setupWithNavController` to have any effect